src/simple-length.js

```diff
@@ -26,6 +26,8 @@
           this.value = nValue;
         }
       }
+    } else {
+      throw new TypeError('\'' + type + '\' is not a valid type for a SimpleLength.');
     }
     if (this.value == undefined) {
       throw new TypeError('Value of SimpleLength must be a number or a numeric string.');
```

Comment:
I thought we were getting rid of the string constructors anyway, so it would end up being something more like

function SimpleLength(value, type) {
if (typeof value != 'number') {
throw "value must be a number"
}
if (shared.LengthValue.LengthType.indexOf(type) < 0) {
throw "invalid argument for unit"
}

this.value = value;
this.type = type;
}

Which I think is much more clear because the validation and the actual setting of the member values is grouped. The check for number of arguments is unnecessary, as extra arguments will be ignored, and if too few are specified, value and/or type will just be undefined.